### ISSUE TITLE
Rotated text from slides.com presentation prints horizontally

### ISSUE THREAD

**Issue by houssemGeek** [astefanutti/decktape#86]
I created a presentation using Reveal (see, please, the screenshot below) that containing a vertical text, but unfortunately this can't appear correctly in my pdf.

![screencapture](https://cloud.githubusercontent.com/assets/6786605/23019416/846da9ea-f442-11e6-9a95-61f74261c7ab.png)

[mypres.pdf](https://github.com/astefanutti/decktape/files/779890/mypres.pdf)

**Comment by astefanutti** [comment]
How is the text vertically rotated? Could you try the following: http://stackoverflow.com/a/32971855/601347

Next major item for DeckTape that I plan is to rebase on top of PhantomJS 2.5 which brings CSS3 support so standard `transform:rotate` will be working.

**Comment by houssemGeek** [comment]
I just roted the text box in the editor that I use (see, please, the screenshot below).
![screen1](https://cloud.githubusercontent.com/assets/6786605/23023733/28c5c3c4-f457-11e6-9fd3-643537be8baf.png)
![screen2](https://cloud.githubusercontent.com/assets/6786605/23023735/2b858284-f457-11e6-854b-b05b8cbf138b.png)

**Comment by astefanutti** [comment]
I've just tried exporting a simple deck from the Reveal / slides.com editor and it confirms the editor does not output the `-webkit-transform` attribute:
```
<div class="reveal">
  <div class="slides">
    <section data-id="...">
      <div class="sl-block" data-block-type="text" style="width: 383px; left: 159px; top: 338px; height: auto;" data-block-id="...">
        <div class="sl-block-style" style="z-index: 0; transform: rotate(270deg);">
          <div class="sl-block-content" data-placeholder-tag="h1" data-placeholder-text="Title Text>
            <h1>Title Text</h1>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
```

A work-around for you is to export your presentation and edit the HTML to add the missing `-webkit-transform` attribute. The long term solution will be for DeckTape to rely on the latest PhantomJS version that brings proper CSS3 support.

**Comment by houssemGeek** [comment]
Thanks, it works for me. Otherwise, I tried to print a screenshot but it doesn't work unfortunately ... an other feature to add?

**Comment by astefanutti** [comment]
> I tried to print a screenshot but it doesn't work unfortunately ...

Do you mean you tried exporting the slides with the `--screenshots` and it didn't work? What error did you face?

**Comment by houssemGeek** [comment]
This the command that I used:
`./phantomjs decktape.js --screenshots  --slides 1 my_url screenshot.png`
And this is the image:
![screenshot](https://cloud.githubusercontent.com/assets/6786605/23061692/2ef480f8-f503-11e6-923d-c8790d9da2d5.png)

**Comment by astefanutti** [comment]
Ah, for the moment the second positional argument is always the PDF document to export. the `--screenshots` option exports the screenshots in the `screenshots` directory (can be changed with the `--screenshots-directory ` option). So you should be doing:

    ./phantomjs decktape.js --screenshots --slides 1 my_url slide.pdf

And you'll get the screenshot written in the `screenshots` directory.

I'm thinking about refactoring it so that you can pass it to the positional arguments instead, but that's low priority at the moment.

**Comment by houssemGeek** [comment]
Great, I founded my image in the `screenshots` directory.

**Comment by houssemGeek** [comment]
astefanutti  I found something that looks strange to me:
I put a static Html code in my view `-webkit-transform: rotate (270deg);` and I print my page correctly 

Otherwise, I diminish the process it doesn't work any more!

`var slBlock = $('.sl-Block');`
      `        slBlock.each(function () {`
   `                    var angle = getRotationDegrees($(this));`
   `                    if (angle == 90) {`
  `                        $(this).css('-webkit-transform', angle);`
  `                        console.log(angle);  // 90 in navigator and in terminal 0 ! ` 
  `                    }`
  `                });`

**Comment by astefanutti** [comment]
Ah that should theoretically work! There must be an error or a small difference somehow.

I'm surprised it prints `0` even with the `angle == 90` condition!

**Comment by houssemGeek** [comment]
hhhh you have reason I mean:
`var slBlock = $('.sl-Block');
slBlock.each(function () {
var angle = getRotationDegrees($(this));
if (angle == 90) {
console.log(angle); // 90 in navigator and in terminal 0 !
$(this).css('-webkit-transform', angle);
 // ...
}
});`

**Comment by astefanutti** [comment]
So it looks like there is a difference in the `getRotationDegrees` method. What's the method code?

**Comment by houssemGeek** [comment]
Ok, this the `getRotationDegrees` function:

`function getRotationDegrees (obj) {
    var matrix = obj.css("-webkit-transform") || obj.css("transform");
    var angle = 0;
    if (matrix !== 'none') {
        var values = matrix.split('(')[1].split(')')[0].split(',');
        var x = values[0];
        var y = values[1];
        angle = Math.round(Math.atan2(y, x) * (180/Math.PI));
    } 

    return angle;
}`

**Comment by houssemGeek** [comment]
Finally, I managed to do it!

```
if (!String.prototype.includes) {
          String.prototype.includes = function (search, start) {
              'use strict';
              if (typeof start !== 'number') {
                  start = 0;
              }

              if (start + search.length > this.length) {
                  return false;
              } else {
                  return this.indexOf(search, start) !== -1;
              }
          };
      }

      var slBlock = jQuery('.sl-block-style');
      if (typeof slBlock !== 'undefined') {
          slBlock.each(function () {
              var _style = jQuery(this).attr('style');
              if (typeof _style !== typeof undefined && _style !== false) {
                  if (_style.includes('transform')) {
                      _style = _style.replace('transform', '-webkit-transform');
                      jQuery(this).attr('style', _style);
                  }
              }
          });
      }
```
What do you think?

**Comment by astefanutti** [comment]
Cool! So that leads to the question whether we want to integrate that kind of adaptation into DeckTape.

It could be done either in the `configure` method of the Reveal plugin, or globally through an extension mechanism, like I drafted a while ago @eb0e3c63c91484b9d350175d19bd07026e68a924, as it's not Reveal specific after all.

On the other hand, it will become irrelevant when DeckTape will be rebased on latest PhantomJS version, so I'm tempted to avoid introducing that extra complexity in the meantime.

**Comment by houssemGeek** [comment]
Great, I am excited for the next PhantomJS!

**Comment by houssemGeek** [comment]
I'am foxing on generating screenshots for my presentation (like above) with `deckatape`, but I have both screenshots and pdf. Can I get only the screenshots to reduce execution time?

**Comment by astefanutti** [comment]
Ah no, for the time being you always get the PDF produced. I'm thinking about refactoring the CLI so that PDF becomes optional. In the meantime, you can work-around by removing the `then(printSlide)` at: https://github.com/astefanutti/decktape/blob/b6f3830ba939541ecc3900035c11fd9d1ee62b65/decktape.js#L315.

Probably the execution time is due to the pause, that you can change at https://github.com/astefanutti/decktape/blob/b6f3830ba939541ecc3900035c11fd9d1ee62b65/decktape.js#L322.

**Comment by houssemGeek** [comment]
OK, I'am agree with the idea that the pdf will be optional when printing images / screenshots. 
On the same logic, If I reduce the pause for printing pdf at: https://github.com/astefanutti/decktape/blob/b6f3830ba939541ecc3900035c11fd9d1ee62b65/decktape.js#L51, that will be less time?

**Comment by astefanutti** [comment]
Indeed, you can control the pause between each slides with the `--pause` option. It defaults to 1s so that most frameworks will complete rendering the slides within that timeframe, though you can reduce that pause in case you know your slides render faster.

**Comment by houssemGeek** [comment]
Cool, I win nearly half time execution in the pdf! but I still have nearly the same time in printing screenshot.
Can I decrease the time at https://github.com/astefanutti/decktape/blob/b6f3830ba939541ecc3900035c11fd9d1ee62b65/decktape.js#L328 , ?

**Comment by astefanutti** [comment]
Yes, that delay serves the same purpose so you can try decreasing it as well.

**Comment by astefanutti** [comment]
Should be fixed in version `2.0.1` that uses headless Chrome. Let me know if you still face this issue.
